### DISCUSSION TITLE
Games in os.js?

### DISCUSSION THREAD

**Discussion by ghost** [os-js/OS.js#817] [Q&A] [answered]
Hi everyone.

#### Intro
I was thinking about creating games for os.js, but... there is a repository called: [KilledByAPixel/OS13k](https://github.com/KilledByAPixel/OS13k). So... there are several games on OS13k, I think there could be OS13k compatibility with os.js, since both are web projects (html, css, js).

#### what is OS13k?
- a tiny OS and mini game engine
- OS13k is a tiny web based pseudo OS and game engine designed for creative coding purposes
- OS13k is a tiny operating system that fits in a 13 kb zip file.
- It includes native support for Shadertoys, Dweets, ZzFX sounds, music, trophies, touch input, and much more.

#### why?
1. I was wondering if there is a possibility to merge 2 projects(OS13k+os.js) into something unique. That way I could develop games for os.js and OS13k at the same time on a single platform. 
2. I recently talked here about the possibility of having a real operating system.  I believe the web games market is a starting point for this.

#### demo?
If you want to play some games you can access this link: [https.os13k.com](https://killedbyapixel.github.io/OS13k/)

#### concept?
![image](https://user-images.githubusercontent.com/111701513/206919919-dc75d519-0d51-4003-9368-da69d81cee17.png)

*Have you all thought about this idea?*

**Comment by andersevenrud** [comment]
> think there could be OS13k compatibility with os.js, since both are web projects (html, css, js).

It depends. In theory you can run *anything* in OS.js depending on the integration and dependencies.

As you said, it's "just web", but if the OS13k apps depends on internal APIs, then you'd have to create some kind of glue to make that work in OS.js or make embedded versions to run in separate iframes.

**Reply by ghost** [reply]
Hi andersevenrud, thank you for feedback.

**Comment by andersevenrud** [comment, marked as answer]
> I recently talked here about the possibility of having a real operating system. 

I haven't been able to look at this thread, but that was never really an objective of this project. It's for *the web*. The main issue with a "real OS" in JS is that you chew away at performance and integration support etc. (It's single threaded and does not have great hardware support in general for rendering pipelines etc with X or Wayland). Which is why for example Gnome Shell allows writing extension in anything you want (even JS), but the core is still C (or something like that).

**Reply by ghost** [reply]
Hi andersevenrud, thank you for feedback.

**Comment by ajmeese7** [comment]
I've actually already implemented games in my fork of OS.js, you can check it out on https://aaronmeese.com or in the [MeeseOS repo](https://github.com/meeseOS/meeseOS).

**Reply by ghost** [reply]
Hi ajmeese7, thank you for feedback.
